### ISSUE TITLE
In some situations orca doesn't read the search settings label when we enter VSCode settings

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#142064]
Issue Type: <b>Bug</b>

CC: @isidorn
CC: @joanmarie

Steps to reproduce:
1. Launch orca.
1. Launch VSCode.
1. Go to preferences:open settings(UI) pressing ctrl+, .

Actual results:
Orca stays mute.
Expected results:
Orca should read: "search settings editor".

1. type for example "status" in the search field.
1. Move to the clear button pressing tab, and press enter or space bar to activate the button.

Actual results:
Focus moves to search field but orca doesn't read anything.
Expected results:
The focus should move to the search field and the orca should read the label of that field.

VS Code version: Code - Insiders 1.64.0-insider (5554b12acf27056905806867f251c859323ff7e9, 2022-02-02T17:16:05.517Z)
OS version: Linux x64 5.16.4-arch1-1
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-5200U CPU @ 2.20GHz (4 x 2344)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 1, 1|
|Memory (System)|15.54GB (1.06GB free)|
|Process Argv|-n --crash-reporter-id bcecf5e0-97d1-4830-b28c-0943c4adab1c --crash-reporter-id bcecf5e0-97d1-4830-b28c-0943c4adab1c|
|Screen Reader|yes|
|VM|0%|
|DESKTOP_SESSION|gnome-xorg|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome-xorg|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (37)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-eslint|dba|2.2.2
EditorConfig|Edi|0.16.4
prettier-vscode|esb|9.2.0
shell-format|fox|7.2.2
rest-client|hum|0.24.6
vscode-docker|ms-|1.19.0
csharp|ms-|1.24.0
vscode-aks-tools|ms-|0.0.11
vscode-kubernetes-tools|ms-|1.3.6
python|ms-|2022.0.1762517795-dev
vscode-pylance|ms-|2022.1.6-pre.1
remote-containers|ms-|0.218.0
remote-ssh|ms-|0.70.0
remote-ssh-edit|ms-|0.70.0
azure-account|ms-|0.9.11
powershell|ms-|2021.12.0
vsliveshare|ms-|1.0.5330
vsliveshare-audio|ms-|0.1.91
oracledevtools|Ora|21.3.0
vscode-boot-dev-pack|Piv|0.1.0
vscode-concourse|Piv|1.30.0
vscode-manifest-yaml|Piv|1.30.0
vscode-spring-boot|Piv|1.30.0
fabric8-analytics|red|0.3.5
java|red|1.3.0
vscode-commons|red|0.0.6
vscode-xml|red|0.18.2
vscode-yaml|red|1.3.0
sonarlint-vscode|Son|3.2.0
vscodeintellicode|Vis|1.2.16
vscode-java-debug|vsc|0.38.0
vscode-java-dependency|vsc|0.19.0
vscode-java-pack|vsc|0.21.0
vscode-java-test|vsc|0.34.0
vscode-maven|vsc|0.35.0
vscode-spring-boot-dashboard|vsc|0.2.0
vscode-spring-initializr|vsc|0.8.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30422396
pythontb:30258533
pythonvspyt551cf:30291413
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
pythonvspyt200:30323110
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30404738
wslgetstartedc:30421358
vscop453:30404998
vsclayoutctrc:30405799
pythonvsnew555:30426298
vscscmwlcmc:30428972

```

</details>

**Comment by isidorn** [comment]
I can not reproduce this with VoiceOver or NVDA. The "Search Settings" always gets read. @rzhao271 can you just double check if you properly set the aria-label on the Preferences input

@joanmarie I believe this is an issue on the Orca side because things are working with VoiceOver and NVDA. Can you double check when you find the time?

**Comment by joanmarie** [comment]
Looks like another case of Orca getting events and silently updating focus. Will see what I can do.

**Comment by joanmarie** [comment]
I've just made in change in Orca master which solves the problem. As with other such changes, there's a risk of introducing unwanted chattiness for badly behaved web apps and browser implementations -- and possibly for well-behaved ones doing something I've not thought about *grins*. Thorough testing for such side effects is strongly encouraged.

**Comment by isidorn** [comment]
Thanks @joanmarie 
Let's see what @jvesouza says and if it is fixed we can close this.
If there are other issues that came up as part of this we can revert the change, I do not think it is critical to fix imho.

**Comment by jvesouza** [comment]
It's working correctly.
However I found another small problem that may or may not be associated with this one.
If I type something in the search field orca correctly announces the number of elements that satisfy the search. If I press the tab key and press enter to activate the clear button, the focus moves to the search field as expected and the orca announces that N items were found, where N is the same number of elements found when I typed something in the search before I press the tab key.

**Comment by isidorn** [comment]
@jvesouza awesome! 
The new issue you found sound like a VS Code bug.
@rzhao271 can you verify if the clear action updates the `aria-label` properly. It seems like the `aria-label` gets stale with old results.

**Comment by isidorn** [comment]
@rzhao271 great, thanks for fixing it this fast 👏 
@jvesouza can you just verify tomorrow and let us know if it is good for you know.

Closing this issue since Raymond pushed a fix.

**Comment by jvesouza** [comment]
I generated a version from the repository, or at least I hope I did, and if I did everything right the problem was fixed. What caught my attention is that the orca is reading more information when I hit the clear button. After the information that the button is a push button, the orca reads something like grade.
It was present in previous versions but I forgot to mention.

**Comment by isidorn** [comment]
@jvesouza great that the problem is fixed -> verified.
As for the reading of "grade" we are not appending this from the VS Code side. Can you share exactly what unnecessary part is read?

**Comment by jvesouza** [comment]
@isidorn  @joanmarie 
When I move the foco to the clear input button orca reads the following:
'Clear Settings Search Input push button grayed.'.
Unless I'm mistaken, the word 'grayde' read by the orca would indicate that the button is disabled, but this word is always read whether or not the button is disabled.

**Comment by isidorn** [comment]
I checked and we properly set `aria-disabled="true"` on an action once it is disabled (grayed). Once the action is enabled we remove that property. Might that be confusing Orca? Should we instead set `aria-disabled="false"` then @joanmarie

**Comment by joanmarie** [comment]
@isidorn: Orca doesn't look at ARIA properties in that case; it looks at accessible states. From what is being described here, setting it to `false` rather removing the property would not have any impact. Instead, it sounds like there is a bug in which toggling `aria-disabled` doesn't result in an accessible state-changed event. As a result, AT-SPI2 tells Orca the widget is still disabled because nothing told AT-SPI2 that had changed so its cached value isn't updated.

If you have a moment (right now I don't), please file a bug against Chromium with concrete steps to reproduce the problem. If you're able to assign the bug (via the `owner` field) please do; otherwise please CC me. In either case, please use my igalia.com email address.

Thanks!

**Comment by jvesouza** [comment]
@isidorn  @joanmarie 
I don't know if this information can help, but it looks like the orca is reading the menus correctly. When for example I have changed files, the 'save all' menu is not read by the orca as 'grayde'. However, if all files have no pending changes, the 'save all' menu is read as 'grayde'.

**Comment by isidorn** [comment]
@jvesouza "Save All" gets greyed out when there are no pending changes. So this is expected behaviour.

As for the initial "Clear" action being read as greyed I can reproduce this with VS Code insiders.
I can not reproduce it with Chrome 98. So it seems like Chrome has fixed this issue in later versions.
@jvesouza can you try going to https://vscode.dev/github/microsoft/vscode that will load up the vscode workbench and try to reproduce there in Chrome browser. It works for me. So it seems we just need to pick a later Chrome version with the fix.

**Comment by jvesouza** [comment]
@isidorn Confirmed.

**Comment by isidorn** [comment]
@jvesouza great, then I suggest no action and we will sooner or later pick this fix up. Especially since it is not a big blocker, I suggest we do not look into back porting it.
